frontend/cs_api_test.go

```diff
@@ -47,29 +61,53 @@ func TestIsBool1(t *testing.T) {
 
 	var circuit IsBool1
 
```

Comment:
these Bool tests should be in internal/backend/circuits with the others. 

Landed here after a bug report few months back. 

Idea is to ensure that constraining linear expressions to be boolean do work as intended, if I recall